# Tugas UTS

# Menyiapkan Data Heart disease

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
# membaca dataset csv dari url
url = 'https://raw.githubusercontent.com/kb22/Heart-Disease-Prediction/master/dataset.csv'
data = pd.read_csv(url)

In [ ]:
data.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

Penjelasan data heart disease

*   age - umur
*   sex - pria=1 wanita=0
*   cp - type sakit dada
*   trestbps - tekanan darah
*   chol - serum cholestoral mg/dl 
*   fbs - gula darah > 120 mg/dl (iya=1 tidak=0)
*   restecg - resting electrocardiographic results
*   thalach - maximum heart rate achieved
*   exang - exercise induced angina (1 = yes; 0 = no)
*   oldpeak - ST depression induced by exercise relative to rest
*   slope - the slope of the peak exercise ST segment
*   ca - number of major vessels (0-3) colored by flourosopy
*   thal - 3 = normal; 6 = fixed defect; 7 = reversable defect
*   target - hasil prediksi sakit=1 tidak=0 



## Implementasi klasifikasi KNN



memisahkan hasil prediksi dengan fitur

In [ ]:
# pilih semua kolom kecuali kolom terakhir
x = data.iloc[:, :-1]
# pilih hanya kolom target
y = data.loc[:, "target"]
y = data['target'].values

bagi data menjadi data train dan data test, 80:20

In [ ]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.2,random_state=1)

import model knn dari sklearn

In [ ]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=5)
knn.fit(x_train,y_train)

KNeighborsClassifier()

coba model prediksi knn dengan data baru

In [ ]:
x_new = [[63,1,3,145,233,1,0,150,0,2.3,0,0,1]]
y_predict = knn.predict(x_new)

print(y_predict[0])

1


/usr/local/lib/python3.9/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but KNeighborsClassifier was fitted with feature names
  warnings.warn(


 ## Implementasi Klasifikasi Naive Bayes

bagi data menjadi data train dan data test

In [ ]:
# membagi data menjadi set train dan test (80:20)
from sklearn.model_selection import train_test_split
x_train2,x_test2,y_train2,y_test2 = train_test_split(x,y,test_size=0.3,random_state=0)

import model naive bayes

In [ ]:
from sklearn.naive_bayes import GaussianNB
nb = GaussianNB()
nb.fit(x_train2, y_train2)

GaussianNB()

prediksi data baru dengan model naive bayes

In [ ]:
x_new2 = [[63,1,3,145,233,1,0,150,0,2.3,0,0,1]]
y_predict2 = nb.predict(x_new2)

print(y_predict2[0])

0


/usr/local/lib/python3.9/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names
  warnings.warn(
